libazureinit/src/media.rs

```diff
@@ -75,18 +77,28 @@ pub const PATH_MOUNT_DEVICE: &str = "/dev/sr0";
 pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 
 const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
+const MTAB_PATH: &str = "/etc/mtab";
 
 // Get a mounted device with any filesystem for CDROM
 #[instrument]
-pub fn get_mount_device() -> Result<Vec<String>, Error> {
-    let mut list_devices: Vec<String> = Vec::new();
+pub fn get_mount_device(path: Option<&Path>) -> Result<Vec<String>, Error> {
+    // Create a new FsTab instance and parse the mtab file
+    let fstab = FsTab::new(path.unwrap_or(Path::new(MTAB_PATH)));
 
-    while let Some(device) = block_utils::get_mounted_devices()?
+    // Get the entries from the FsTab
```

Comment:
I would drop these comments (create new fstab, get the entries...) as they don't really add any meaningful value. Anyone reading the code can already tell that the method will "get the entries from the FsTab"